### PR TITLE
add documentation of how to select components

### LINKED ISSUE #101: Select which kinds of components to return
Before v1.1.0, the library returned only events. Now, it is also returning other component types, see #97.
I am unsure if my code breaks when I update! https://github.com/niccokunzmann/open-web-calendar/pull/136 (yep, not everything is TDD with me :)

I would like to propose a way to
- by default only return events
- additionally add components that we also like

```
events = recurring_ical_events.of(calendar).between(...)
journals = recurring_ical_events.of(calendar, components=["VJOURNAL"]).between(...)
todos = recurring_ical_events.of(calendar, components=["VTODO"]).between(...)
all = recurring_ical_events.of(calendar, components=["VTODO", "VEVENT", "VJOURNAL"]).between(...)
```

It almost looks like we should do

```
events = recurring_ical.events.of(calendar).between(...)
todos = recurring_ical.todos.of(calendar).between(...)
journals = recurring_ical.journals.of(calendar).between(...)
```
But I can see that you might want them all in one place - so this might be not the right design. Renaming the library to `recurring_ical_components` might also be a nice step.

@tobixen this is interesting to you.

### PR DESCRIPTION
**Pull request by niccokunzmann**
fixes #101

- [x] add documentation
- [x] add tests
- [x] fix tests
- release

### CODE REVIEW

**Comment by tobixen** [comment]
Should have looked better into this in November already ... but there is so much that needs to be done and so little time for everything.

It seems that by now I have to replace a one-liner with three code lines to perform the same thing - and it's not much intuitive how to toggle between only expanding events and expanding everything.  I think we should consider an easier interface for "expand everything".

**Comment by niccokunzmann** [comment]
> It seems that by now I have to replace a one-liner with three code lines to perform the same thing - and it's not much intuitive how to toggle between only expanding events and expanding everything. I think we should consider an easier interface for "expand everything"..

Hm easy sounds good! I wonder what people expect. If you have expectations, please open an issue and tell me how you would like to use the library. I use it for the open-web-calendar and for me, this is fine like this.

Also, we are still missing alarms and time zones in that and there might be other components.

**Comment by tobixen** [comment]
tldr-summary: I would expect either the 1.1.0b-behavior (where all event/task/journals are expanded) or I would expect a parameter that is easy to set (like, something that can be passed to the `of` or `between` methods).  I will try to find time to think a bit more about this and come with a suggestion.

----

To be honest, my expectation is the 1.1.0b-behavior, where all event/task/journals are expanded without explicitly asking for it.  Even if I can "fix" my caldav library to specifically ask for all event/task/journals to be expanded, I can't fix it on the server side for all servers that may be running python and depend on the recurring-ical-event library.

Events, tasks and journals are very similar in nature, they share mostly the same attributes.  A task is an event with a flexible start, and a journal is an event that has already happened.  At the other hand, alarms and time zones are something else - they have quite different attributes.  Alarms can arguably be considered to be "events telling that other events are about to happen", but anyway, in the RFCs they are a very different thing.  Different code would be needed to expand it as well, as it doesn't have the rrule property.

Recurring tasks are found everywhere if one cares looking for them - like house work chores, checking the mailbox.  There is the example from RFC5545 - "Submit Quebec Income Tax Return 2006" - admittedly the original comes without RRULE, but delivering tax documents is a typical recurrent task.  I have some similar administrative burden that I need to do at the 1st of the month, it's clearly a task), and it's clearly recurrent. 

Anyway, I'm not insisting that the 1.1.0b-behavior is the only right thing.  I've said earlier that expanding of  recurring tasks may be a bit obscure (and for journals - even more obscure, I can't see many use cases for that - though they may exist).  For tasks that needs to be done on the first of the month or every Monday, expanding may be useful, I would wish to see those things if scheduling some activity for some future date - but at the other hand, for most tasks there is no strong relationship between the due-timestamp and the timestamp it's actually performed, so it may probably not be a good idea to list up arbitrary tasks when checking up the agenda for some future date.

I chose not to argue when you made the decision to add a parameter, but I didn't realize the parameter was to an `__init__`  that one doesn't explicitly call during normal code execution.  To make it easy to pass the parameter, it should be accepted either by the `of`-method or `between`-method.

**Comment by tobixen** [comment]
Nevermind ... I was just copying  code from test, didn't have the full understanding of how the `calendars`-argument was generated.  I see now that I can pass the `components`-parameter to the `of`-method, so that suffices for me.

(I still think the default should be to expand all task-alike components, but at the other hand it would be plain silly to change the behavior in 2.0 and then revert this and have to release a new 3.0 right afterwards  ...) :-)

**Comment by niccokunzmann** [comment]
Good that you found that parameter. I wondered also, my thought was that other people want to select what to repeat. The value names are different in the components and the components are handeled differently. Hm. It might not matter how we do it. There seems no clear case, just thinking what might be in my head.
